lib/core/presentation/widgets/products_box.dart

```diff
@@ -19,12 +21,30 @@ class ProductsBox extends StatefulWidget {
 }
 
 class _ProductsBoxState extends State<ProductsBox> {
-  bool isClicked = false;
+  bool isFavoriteProduct = false;
+  FavouritesViewModel favouritesViewModel = FavouritesViewModel();
+  StorageService storageService = StorageService();
 
-  void saveItem() {
-    setState(() {
-      isClicked = !isClicked;
-    });
+  Future<void> checkForProduct() async {
+    isFavoriteProduct = await storageService.doesProductExist(widget.product);
+    setState(() {});
+  }
+
+  void saveItem(Product product) {
+    if (isFavoriteProduct == true) {
```

Comment:
`isFavoriteProduct` is already a boolean so you don't need to compare it to true.

The code below does the same thing

```dart
    if (isFavoriteProduct) {
        ....
```